node/overseer/src/metrics.rs

```diff
@@ -40,7 +38,10 @@ struct MetricsInner {
 	signals_sent: prometheus::GaugeVec<prometheus::U64>,
 	signals_received: prometheus::GaugeVec<prometheus::U64>,
 
+	// Allow unused code as they might be enabled only for the jemalloc-stats feature flag.
+	#[allow(dead_code)]
 	memory_stats_resident: prometheus::Gauge<prometheus::U64>,
+	#[allow(dead_code)]
```

Comment:
I think these dead_code allows can be avoided too. 🙂

Reply:
When they are not available, we they should not be registered in prometheus.  Aka also put a cfg above them

Reply:
> When they are not available, we they should not be registered in prometheus. Aka also put a cfg above them

What is the harm in registering unavailable metrics, if we never call them? (I believe you, I actually don't know the answer.)

PVF has some metrics that would need this treatment: `preparation_max_rss`, `preparation_max_allocated`, `preparation_max_resident`. I can take care of that, but honestly was hoping to get away without ugly `#[cfg]` directives everywhere. 🙃

Reply:
> What is the harm in registering unavailable metrics, if we never call them? (I believe you, I actually don't know the answer.)

There is no real harm, just people that will ask why the metric is zero :P Metric being not there is IMO better than just not working.